spirv_msl.cpp

```diff
@@ -398,6 +510,19 @@ void CompilerMSL::emit_header()
 	statement("");
 	statement("using namespace metal;");
 	statement("");
+    emit_msl_defines();
+}
+
+void CompilerMSL::emit_msl_defines()
+{
+    statement("// Standard GLSL->MSL redefinitions");
+    statement("#define discard discard_fragment()");
```

Comment:
These are remapping defines, not ops passed in from SPIRV. Not clear how you think emit_instruction will be useful here. Can you elaborate?